src/internal/connector/onedrive/data_collections_test.go

```diff
@@ -0,0 +1,113 @@
+package onedrive
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+type DataCollectionsUnitSuite struct {
+	tester.Suite
+}
+
+func TestDataCollectionsUnitSuite(t *testing.T) {
+	suite.Run(t, &DataCollectionsUnitSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *DataCollectionsUnitSuite) TestMigrationCollections() {
+	u := selectors.Selector{}
+	u = u.SetDiscreteOwnerIDName("i", "n")
+
+	od := path.OneDriveService.String()
+	fc := path.FilesCategory.String()
+
+	type migr struct {
+		full string
+		prev string
+	}
+
+	table := []struct {
+		name            string
+		version         int
+		expectLen       int
+		expectMigration []migr
+	}{
+		{
+			name:            "zero version",
+			version:         0,
+			expectLen:       0,
+			expectMigration: []migr{},
+		},
+		{
+			name:            "above current version",
+			version:         version.Backup + 5,
+			expectLen:       0,
+			expectMigration: []migr{},
+		},
+		{
+			name:      "user pn to id",
+			version:   version.OneDrive7MigrateUserPNToID - 1,
+			expectLen: 1,
+			expectMigration: []migr{
+				{
+					full: strings.Join([]string{"t", od, "i", fc}, "/"),
+					prev: strings.Join([]string{"t", od, "n", fc}, "/"),
+				},
+			},
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			mc, err := migrationCollections(nil, test.version, "t", u, nil, control.Options{})
+			require.NoError(t, err, clues.ToCore(err))
+
+			if test.expectLen == 0 {
+				assert.Nil(t, mc)
+				return
+			}
+
+			assert.LessOrEqual(t, test.expectLen, len(mc))
+
+			migrs := make([]bool, len(test.expectMigration))
+
+			for _, col := range mc {
+				oc, ok := col.(*Collection)
+				require.True(t, ok, "casting backupCollection iface to *onedrive.Collection")
+
+				var fp, pp string
+
+				if oc.FullPath() != nil {
+					fp = oc.FullPath().String()
+				}
+
+				if oc.prevPath != nil {
+					pp = oc.prevPath.String()
+				}
+
+				t.Logf("Found migration collection:\n* full: %s\n* prev: %s\n", fp, pp)
```

Comment:
can be removed?

Reply:
Nah, keep it.  Otherwise we don't list what collections we did find, which makes this difficult to debug.